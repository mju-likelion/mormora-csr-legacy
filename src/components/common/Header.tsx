import styled from '@emotion/styled';
import { useState } from 'react';
import { Link } from 'react-router-dom';

import ModalPortal from 'ModalPortal';
import Button from 'components/base/Button';
import Responsive from 'components/base/Responsive';
import LoginModal from 'components/modal/LoginModal';
import SigninModal from 'components/modal/SigninModal';

const Block = styled.header`
  position: sticky;
  top: 0;
  left: 0;
  background-color: #212121;
  color: #fafafa;
  width: 100%;
`;

const Inner = styled(Responsive)`
  display: flex;
  justify-content: space-between;
  align-items: center;
  list-style: none;
  height: ${({ theme }) => theme.constants.headerHeight};
`;

const Left = styled.div``;

const Logo = styled.img`
  /* align-self는 safari에서 깨지는거 방지용 */
  align-self: center;
  width: ${({ theme }) => theme.constants.logoWidth};
  height: auto;
`;

const Right = styled.div``;

const LoginButton = styled(Button)`
  padding: 0.5rem 1rem;
`;

const RegisterButton = styled(LoginButton)`
  border: 0.75px solid gray;
  border-radius: 8px;
`;

function Header() {
  const [loginModalOpen, setLoginModalOpen] = useState(false);
  const [signinModalOpen, setSigninModalOpen] = useState(false);

  function toggleLoginOpen() {
    setLoginModalOpen(prev => !prev);
  }

  function toggleSigninOpen() {
    setSigninModalOpen(prev => !prev);
  }

  return (
    <>
      <Block>
        <Inner>
          <Left>
            <Link to="/">
              <Logo
                src="https://mju-likelion.s3.ap-northeast-2.amazonaws.com/static/basiclogo_E_H_W.png"
                alt="MJU Likelion Logo"
              />
            </Link>
          </Left>
          <Right>
            <LoginButton onClick={toggleLoginOpen}>로그인</LoginButton>
            <RegisterButton onClick={toggleSigninOpen}>회원가입</RegisterButton>
          </Right>
        </Inner>
      </Block>
      <ModalPortal>
        {loginModalOpen && <LoginModal onClose={toggleLoginOpen} />}
        {signinModalOpen && <SigninModal onClose={toggleSigninOpen} />}
      </ModalPortal>
    </>
  );
}

export default Header;
