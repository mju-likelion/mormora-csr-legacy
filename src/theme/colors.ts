export const likelionOrange = '#f39926';
export const likelionBlack = '#0d0a0a';
