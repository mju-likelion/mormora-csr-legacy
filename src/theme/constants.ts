export const headerHeight = '4rem';
export const logoWidth = '9rem';
