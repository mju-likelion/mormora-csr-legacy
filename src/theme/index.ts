import * as constants from './constants';

const theme = { constants };

export default theme;

export type Theme = typeof theme;
