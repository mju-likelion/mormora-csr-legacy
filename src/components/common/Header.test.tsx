import { ThemeProvider } from '@emotion/react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import theme from 'theme';

import Header from './Header';

describe('<Header />', () => {
  it('matches snapshot', () => {
    const header = render(
      <BrowserRouter>
        <ThemeProvider theme={theme}>
          <Header />
        </ThemeProvider>
      </BrowserRouter>,
    );

    expect(header.container).toMatchSnapshot();
  });

  it('has a logo', () => {
    const header = render(
      <BrowserRouter>
        <ThemeProvider theme={theme}>
          <Header />
        </ThemeProvider>
      </BrowserRouter>,
    );

    const logo = header.getByAltText('MJU Likelion Logo');
    expect(logo).toBeInTheDocument();
  });
});
