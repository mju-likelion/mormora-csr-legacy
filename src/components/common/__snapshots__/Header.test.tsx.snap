// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<div>
  <header
    class="css-1eal7hw"
  >
    <ul
      class="css-lqika9"
    >
      <li
        class="css-0"
      >
        <a
          href="/"
        >
          <img
            alt="MJU Likelion Logo"
            class="css-1f0ylk0"
            src="https://mju-likelion.s3.ap-northeast-2.amazonaws.com/static/basiclogo_E_H_W.png"
          />
        </a>
      </li>
    </ul>
  </header>
</div>
`;
